### DISCUSSION TITLE
Immich LXC without Docker

### DISCUSSION THREAD

**Discussion by JaviLerma** [tteck/Proxmox#3482] [Ideas]
Maybe works
https://github.com/loeeeee/immich-in-lxc
Thanks!

**Comment by divemasterjm** [comment]
Ive tried but there were errors installing if you follow the guide

**Reply by loeeeee** [reply]
I ironed out some issues now. 😃

**Comment by rlimaeco** [comment]
I use this repo and works great: https://github.com/arter97/immich-native

**Reply by Jmanko16** [reply]
I've tried this and it works, but I'm having issues having external libraries work.

I have my synology NAS mounted as NFS share and bind mounted to the LXC, but when adding "external libraries" in immich it doesn't see the photos.  Any ideas?

**Comment by rlimaeco** [comment]
take a look how to mount a directory for proxmox LXC, you need to have this
external disk connected to your proxmox and have some other permissions to
configure, if you have any trouble send me an email

Em qui., 5 de set. de 2024 às 02:39, Jmanko16 ***@***.***>
escreveu:

**Reply by Jmanko16** [reply]
I have it mounted to proxmox and bind mounted to the lxc.  It is accessible and working from command line.  In immich it is not accessible.  I would assume I need to somehow change the script for install to point to the folder, but I'm not sure how I would adjust the docker compose file if I used the script to install the LXC.

**Comment by rlimaeco** [comment]
so if the permissions are all right, you have access to all the files with command line so maybe you forgot to add to the Immich double check the path of your files in https://your-immich-url/admin/library-management  the paths the you have access that have all your pictures are there ?
________________________________
De: Jmanko16 ***@***.***>
Enviado: quinta-feira, 5 de setembro de 2024 13:34
Para: tteck/Proxmox ***@***.***>
Cc: Rafael Lima ***@***.***>; Comment ***@***.***>
Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482)


I have it mounted to proxmox and bind mounted to the lxc. It is accessible and working from command line. In immich it is not accessible. I would assume I need to somehow change the script for install to point to the folder, but I'm not sure how I would adjust the docker compose file if I used the script to install the LXC.

**Reply by Jmanko16** [reply]
When I go to library management and add the folder it says it's not accessible.

this is why I am wondering how the script works, aka do I need to tell immich somehow where this folder is? (I would think the whole external library implies I'm adding it after immich is installed but it doesn't show up)

**Comment by rlimaeco** [comment]
Just to be sure, try to change again the permission in your path in the proxmox shell:

chown -R 1005:1005 /mnt/pve/your-local-storage-name

if you have your-local-storage-name mounted at /mnt/pve


I remember that I have to do that, and this is true if you have something like in the following code in your LXC conf:


lxc.idmap: u 0 100000 1005
lxc.idmap: g 0 100000 1005
lxc.idmap: u 1005 1005 1
lxc.idmap: g 1005 1005 1
lxc.idmap: u 1006 101006 64530
lxc.idmap: g 1006 101006 64530

In the same file be sure to have also:

mp0: /mnt/pve/your-local-storage-name,mp=/mnt/your-local-storage-name

I am running with this configuration since last year without any issue with that, I wish you good luck too hehehe
________________________________
De: Jmanko16 ***@***.***>
Enviado: quinta-feira, 5 de setembro de 2024 20:17
Para: tteck/Proxmox ***@***.***>
Cc: Rafael Lima ***@***.***>; Comment ***@***.***>
Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482)


When I go to library management and add the folder it says it's not accessible.

this is why I am wondering how the script works, aka do I need to tell immich somehow where this folder is? (I would think the whole external library implies I'm adding it after immich is installed but it doesn't show up)

**Comment by Mati-l33t** [comment]
Really wants tteck to make a script for this. A script that would ask for the path to the files would be even better 😊

**Comment by rlimaeco** [comment]
I really want too
I am not glad about the way I am using now , now I have to update with a script, and I need to expose the port using socat because I could not figure out how to make Immich use 0.0.0.0 instead of localhost.
Maybe I should take a look how to create a PR for this and maybe tteck get interested too
________________________________
De: Mati ***@***.***>
Enviado: sexta-feira, 6 de setembro de 2024 00:35
Para: tteck/Proxmox ***@***.***>
Cc: Rafael Lima ***@***.***>; Comment ***@***.***>
Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482)


Really wants tteck to make a script for this. A script that would ask for the path to the files would be even better 😊

**Reply by loeeeee** [reply]
> I really want too I am not glad about the way I am using now , now I have to update with a script, and I need to expose the port using socat because I could not figure out how to make Immich use 0.0.0.0 instead of localhost. Maybe I should take a look how to create a PR for this and maybe tteck get interested too
> […](#)
> ________________________________ De: Mati ***@***.***> Enviado: sexta-feira, 6 de setembro de 2024 00:35 Para: tteck/Proxmox ***@***.***> Cc: Rafael Lima ***@***.***>; Comment ***@***.***> Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482) Really wants tteck to make a script for this. A script that would ask for the path to the files would be even better 😊 — Reply to this email directly, view it on GitHub<[#3482 (comment)](https://github.com/tteck/Proxmox/discussions/3482#discussioncomment-10562743)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAJJEXLQJYFAIHNWYIACC6LZVDMEFAVCNFSM6AAAAABMDN5S7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJWGI3TIMY>. You are receiving this because you commented.Message ID: ***@***.***>

Wait... Isn't immich default to using 0.0.0.0? In immich-native you could try to specify the licensing address, `IMMICH_HOST`, specified in https://immich.app/docs/install/environment-variables/ 

You could also run immich API in foreground to see if it is using 0.0.0.0:3001 by default. Its logging is pretty good. 😄

**Comment by rlimaeco** [comment]
I tried once without success, I will take a lot again today after work, thanks for the suggestions ! 🙂
________________________________
De: Loe ***@***.***>
Enviado: sexta-feira, 6 de setembro de 2024 06:14
Para: tteck/Proxmox ***@***.***>
Cc: Rafael Lima ***@***.***>; Comment ***@***.***>
Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482)


I really want too I am not glad about the way I am using now , now I have to update with a script, and I need to expose the port using socat because I could not figure out how to make Immich use 0.0.0.0 instead of localhost. Maybe I should take a look how to create a PR for this and maybe tteck get interested too
…
________________________________ De: Mati @.> Enviado: sexta-feira, 6 de setembro de 2024 00:35 Para: tteck/Proxmox @.> Cc: Rafael Lima @.>; Comment @.> Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482<https://github.com/tteck/Proxmox/discussions/3482>) Really wants tteck to make a script for this. A script that would ask for the path to the files would be even better 😊 — Reply to this email directly, view it on GitHub<#3482 (comment)<https://github.com/tteck/Proxmox/discussions/3482#discussioncomment-10562743>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AAJJEXLQJYFAIHNWYIACC6LZVDMEFAVCNFSM6AAAAABMDN5S7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJWGI3TIMY. You are receiving this because you commented.Message ID: @.***>

Wait... Isn't immich default to using 0.0.0.0? In immich-native you could try to specify the licensing address, IMMICH_HOST, specified in https://immich.app/docs/install/environment-variables/

You could also run immich API in foreground to see if it is using 0.0.0.0:3001 by default. Its logging is pretty good. 😄

**Comment by rlimaeco** [comment]
I tried to set this IMMICH_HOST to 0.0.0.0 (and in the docs is really the default) but I still getting this even after reinstall the immich-native:

Immich Server is listening on http://127.0.0.1:3001 [v1.113.1] [PRODUCTION]

Maybe I made some mistake, but to use I still need socat
________________________________
De: Loe ***@***.***>
Enviado: sexta-feira, 6 de setembro de 2024 06:14
Para: tteck/Proxmox ***@***.***>
Cc: Rafael Lima ***@***.***>; Comment ***@***.***>
Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482)


I really want too I am not glad about the way I am using now , now I have to update with a script, and I need to expose the port using socat because I could not figure out how to make Immich use 0.0.0.0 instead of localhost. Maybe I should take a look how to create a PR for this and maybe tteck get interested too
…
________________________________ De: Mati @.> Enviado: sexta-feira, 6 de setembro de 2024 00:35 Para: tteck/Proxmox @.> Cc: Rafael Lima @.>; Comment @.> Assunto: Re: [tteck/Proxmox] Immich LXC without Docker (Discussion #3482<https://github.com/tteck/Proxmox/discussions/3482>) Really wants tteck to make a script for this. A script that would ask for the path to the files would be even better 😊 — Reply to this email directly, view it on GitHub<#3482 (comment)<https://github.com/tteck/Proxmox/discussions/3482#discussioncomment-10562743>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AAJJEXLQJYFAIHNWYIACC6LZVDMEFAVCNFSM6AAAAABMDN5S7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJWGI3TIMY. You are receiving this because you commented.Message ID: @.***>

Wait... Isn't immich default to using 0.0.0.0? In immich-native you could try to specify the licensing address, IMMICH_HOST, specified in https://immich.app/docs/install/environment-variables/

You could also run immich API in foreground to see if it is using 0.0.0.0:3001 by default. Its logging is pretty good. 😄
